source/overloading.tex

```diff
@@ -3657,60 +3649,42 @@
 \indextext{overloading!function call operator}
 
 \pnum
-\tcode{operator()}
-shall be a non-static member function with an arbitrary number of
-parameters.
-It can have default arguments.
-It implements the function call syntax
+A \defnadj{function call}{operator function}
+is a function named \tcode{\keyword{operator}()}
+that is a non-static member function with an arbitrary number of parameters.
+It may have default arguments.
+For an expression of the form
 \begin{ncsimplebnf}
 postfix-expression \terminal{(} \opt{expression-list} \terminal{)}
 \end{ncsimplebnf}
-where the
-\grammarterm{postfix-expression}
-evaluates to a class object and the possibly empty
-\grammarterm{expression-list}
-matches the parameter list of an
-\tcode{operator()}
-member function of the class.
-Thus, a call
-\tcode{x(arg1,...)}
-is interpreted as
-\tcode{x.op\-er\-a\-tor()(arg1, ...)}
-for a class object
-\tcode{x}
-of type
-\tcode{T}
-if
-\tcode{T::operator()(T1,}
-\tcode{T2,}
-\tcode{T3)}
-exists and if the operator is selected as the best match function by
-the overload resolution mechanism\iref{over.match.best}.
+where the \grammarterm{postfix-expression} evaluates to a class object,
+the operator function
+is selected by overload resolution\iref{over.match.oper},
+and the expression is interpreted as
+\begin{ncsimplebnf}
+postfix-expression \terminal{.} \keyword{operator} \terminal{(}\terminal{)} \terminal{(} \opt{expression-list} \terminal{)}
+\end{ncsimplebnf}
 
 \rSec2[over.sub]{Subscripting}%
 \indextext{subscripting operator!overloaded}%
 \indextext{overloading!subscripting operator}
 
 \pnum
-\tcode{operator[]}
-shall be a non-static member function with exactly one parameter.
-It implements the subscripting syntax
+A \defnadj{subscripting}{operator function}
+is a function named \tcode{\keyword{operator}[]}
+that is a non-static member function with exactly one parameter.
+For an expression of the form
 \begin{ncsimplebnf}
 postfix-expression \terminal{[} expr-or-braced-init-list \terminal{]}
 \end{ncsimplebnf}
+the operator function
+is selected by overload resolution\iref{over.match.oper},
+and the expression is interpreted as
+\begin{ncsimplebnf}
+postfix-expression . \keyword{operator} \terminal{[}\terminal{]} \terminal{(} expr-or-braced-init-list \terminal{)}
+\end{ncsimplebnf}
```

Comment:
This is wrong if we select a built-in candidate (the old wording was correct).

Reply:
Added "If a member function is selected,", in line with the rest.